aries_vcx/src/handlers/out_of_band/receiver.rs

```diff
@@ -133,12 +133,12 @@ impl OutOfBandReceiver {
         Ok(None)
     }
 
-    pub async fn build_connection(&self, agency_client: &AgencyClient, did_doc: DidDoc, autohop_enabled: bool) -> VcxResult<Connection> {
+    pub async fn build_connection(&self, agency_client: &AgencyClient, did_doc: DidDoc, autohop_enabled: bool) -> VcxResult<MediatedConnection> {
```

Comment:
Same thoughts here, we could remove this and let the upper layer to choose to build `MediatedConnection` or `Connection` (or perhaps "Connection" but using did-exchange protocol etc.).